cli/internal/prune/prune.go

```diff
@@ -140,12 +175,18 @@ func (p *prune) prune(opts *opts) error {
 		}
 	}
 	workspaces := []turbopath.AnchoredSystemPath{}
-	targets := []interface{}{opts.scope}
-	internalDeps, err := ctx.TopologicalGraph.Ancestors(opts.scope)
-	if err != nil {
-		return errors.Wrap(err, "could find traverse the dependency graph to find topological dependencies")
+
+	targets := filteredPkgs.Copy()
+	for _, pkg := range filteredPkgs {
+		internalDeps, err := ctx.TopologicalGraph.Ancestors(pkg)
```

Comment:
We should try to avoid walking the same part of the graph multiple times. I think adding a helper method onto `Context` named `TopologicalGraphAncestors` that takes a slice of vertexes as starting points would be useful to add. If you look at the source for [Ancestors](https://github.com/pyr-sh/dag/blob/v1.0.0/dag.go#L30) it's straight forward to expand this to multiple vertexes.

Reply:
Thanks for the link, added that in. The method I added returns both the ancestors and the original nodes, which made things simpler for this use case. Think this will get used elsewhere where that wouldn't be desirable? If so I can either rename or change the behavior.